app/services/data_exports/create_service.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+
+module DataExports
+  # Service used to Create Data Export
+  class CreateService < BaseService
+    DataExportCreateError = Class.new(StandardError)
+    def initialize(user = nil, params = {})
+      super(user, params)
+    end
+
+    def execute
+      @data_export = DataExport.new(params)
+
+      validate_params
+
+      validate_sample_export if @data_export.export_type == 'sample'
+
+      @data_export.user = current_user
+      @data_export.status = 'processing'
+
+      @data_export.save
+
+      return true if @data_export.valid?
+
+      false
+    rescue DataExports::CreateService::DataExportCreateError => e
+      @data_export.errors.add(:base, e.message)
+      false
+    end
+
+    private
+
+    # export_type and export_parameters[ids] are required for data_exports
+    def validate_params
+      unless params.key?('export_type') && params.key?('export_parameters') && params['export_parameters'].key?('ids')
+        raise DataExportCreateError, I18n.t('services.data_exports.create.missing_required_parameters')
+      end
+    end
+
+    # Find the project_ids for each sample, and search/validate the unique set of ids to ensure user has authorization
+    # to export the chosen samples' data
+    def validate_sample_export
+      project_ids = []
+
+      params['export_parameters']['ids'].each do |sample_id|
+        sample = Sample.find_by(id: sample_id)
+        raise DataExportCreateError, I18n.t('services.data_exports.create.invalid_sample_id') if sample.nil?
+
+        project_id = sample.project_id
+        project_ids << project_id unless project_ids.include?(project_id)
```

Comment:
This is more of a question and asking confirmation about the permissions...
I think you might be the first to add checkboxes to the group samples table. 
For the project samples table, you only need to lookup the project once as all the samples belong to the project. We check the direct membership of the project. 
I would have thought that we would want to do the same for group samples. Check the direct membership of the group for the group samples table and add that functionaly to group_policy. Instead, I see you're iterating through all the sample projects. 
If we were to add the checkboxes to the group samples table, we would have to authorize each row individually. 
I am also working on a PR that requires group sample selection. I'm selfishly wondering if @deepsidhu85 or @ericenns can confirm this is how they want to see group samples permissions? If so, I need to update my PR like this one. Sorry and thanks!

Reply:
The `DataExports::CreateService` doesn't have a concept of `Group` or `Project` for selection, so it needs to verify for each sample that the user has the correct permissions. This is independent of displaying checkboxes on the respective samples tables, as this service could be used by the API for example so we have to revalidate permissions. Otherwise a bad user could craft an HTTP request and bypass permissions if we didn't have checking also at the service level.

For the Group Samples table, we don't have to check permissions for each `Sample` we can just check the users permission at the `Group` level.

Reply:
Oh ok. I think that makes sense. I will try to do that same for the group samples workflow submission.